src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponSlice.lua

```diff
@@ -0,0 +1,34 @@
+--!strict
+local Reflex = require("Packages/reflex")
+local Immut = require("Packages/immut")
+local Table = require("src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponTables")
+
+export type BrickbattleWeaponStateActions = {
+    setStatState: (playerName: string, numb: number, stat: string) -> (),
+    removeStatState: (playerName: string, stat: string) -> (),
+}
+
+type Data = {
+    [string]: Table.BrickbattleWeaponTable
+}
+local initialState: Data = {}
+local BrickbattleWeaponsStateSlice = Reflex.createProducer(initialState, {
+    setStatState = function(state, playerName: string, numb: number, stat: string)
+		return Immut.produce(state, function(draft) 
+            local playerData = draft[playerName]
```

Comment:
Another nit: should we use userId to account for player name changes mid game? Unsure if Roblox auto-boots players when that happens